APE14.rst

```diff
@@ -0,0 +1,484 @@
+            kinds of world coordinates. Furthermore, we recommend that the
+            output be kept as human-readable as possible.
+            """
+
+We now take a look at an example of use of ``world_axis_object_components`` with
+``world_axis_object_classes``. An example output from both methods on the same
+WCS object is:
+
+.. code-block:: python
+
+    >>> wcs.world_axis_object_components
+    [('skycoord', 'ra'), ('time', 0), ('skycoord', 'dec')]
+    >>> wcs.world_axis_object_classes
+    {'skycoord': ('astropy.coordinates.SkyCoord',
+                  {'frame': 'fk5', 'equinox':'J2005'}),
+     'time': ('astropy.time.Time', {'scale': 'tai'})}
+
+This indicates that the first and third world axis can be used to instantiate an
+Astropy ``SkyCoord`` object with ``ra=`` set to the first world axis, and
+``dec=`` set to the third axis, and the ``frame=fk5`` and ``equinox=J2005``
+arguments, while the second world axis can be used to instantiate an Astropy
+``Time`` object as the first positional argument, and with the ``scale=tai``
+keyword argument. Note that the coordinate frame classes could be custom
+sub-classes if needed.
+
+Low-level API examples
+^^^^^^^^^^^^^^^^^^^^^^
+
+**Simple 1D spectrum** - a 1D mapping from pixel to wavelength:
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True]]
+    wcs.world_axis_units = ['angstrom']
+    wcs.world_axis_physical_type = ['spectral']
+    wcs.world_axis_object_components = [('spec', 0)]
+    wcs.world_axis_object_classes  = {'spec':('astropy.units.Wavelength':
+                                              {'airorvacwl': 'air'})}
+
+**Simple 2D image mapping** where the pixel axes are lined up with RA and Dec
+(in FITS-WCS this would be CAR)
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True, False], [False, True]]
+    wcs.world_axis_units = ['deg', 'deg']
+    wcs.world_axis_physical_type = ['pos.eq.ra', 'pos.eq.dec']
+    wcs.world_axis_object_components = [(('sc', 'ra'), ('sc', 'dec')]
+    wcs.world_axis_object_classes  = {'sc':('astropy.coordinates.SkyCoord',
+                                {'frame': 'icrs'})}
+
+**Extremely complex spectral data cube** with 3 *pixel* dimensions and 4 *world*
+dimensions. The first two *pixel* dimensions encode a mixed set of spatial
+dimensions and a third dimension which is completely spectral (i.e., the output of
+an IFU detector), and the third *pixel* dimension is a separable fourth world
+dimension encoding time-of-observation.
+
+.. code-block:: python
+
+    wcs.axis_correlation_matrix = [[True, True, False],
+                                   [True, True, False],
+                                   [True, True, False],
+                                   [False, False, True]]
+    wcs.world_axis_units = ['deg', 'deg', 'angstrom', 'day']
+    wcs.world_axis_physical_type = ['pos.galactic.lon', 'pos.galactic.lat', 'em.freq', 'time']
```

Comment:
Tiny typo: I think 'em.freq' should be 'em.wl' here? as you use angstrom and astropy.units.Wavelength.